src/hooks/cashout/usePrecalculatedCashouts.ts

```diff
@@ -0,0 +1,82 @@
+import { type Selection, GraphBetStatus } from '@azuro-org/toolkit'
+import { useMemo } from 'react'
+import { useQuery } from '@tanstack/react-query'
+
+import { useChain } from '../../contexts/chain'
+import { batchFetchCashouts } from '../../helpers/batchFetchCashouts'
+
+
+type Props = {
+  selections: Selection[]
+  graphBetStatus: GraphBetStatus
+  enabled?: boolean
+}
+
+export type PrecalculatedCashout = {
+  isAvailable: boolean
+  multiplier: string
+} & Omit<Selection, 'coreAddress'>
+
+
+export const usePrecalculatedCashouts = ({ selections, graphBetStatus, enabled = true }: Props) => {
+  const { appChain, api, contracts } = useChain()
+
+  const isLive = selections[0]!.coreAddress === contracts.liveCore?.address
+  const conditionsKey = useMemo(() => selections.map(({ conditionId }) => conditionId).join('-'), [ selections ])
+
+  const queryFn = async () => {
+    const data = await batchFetchCashouts(conditionsKey.split('-'), appChain.id)
+
+    const newCashouts = selections.reduce<Record<string, PrecalculatedCashout>>((acc, { conditionId, outcomeId }) => {
+      const key = `${conditionId}-${outcomeId}`
+      const cashout = data?.[key]!
+
+      acc[key] = cashout
+
+      return acc
+    }, {})
+
+    return newCashouts
+  }
+
+  const { data: cashouts, isFetching } = useQuery({
+    queryKey: [ 'cashout/precalculate', api, conditionsKey ],
```

Comment:
you don't use `api` but use `appChain.id`, (it looks the same change but I think it's clearer)

Reply:
not really, api could be the same for example in production chains